election/views.py

```diff
@@ -224,3 +225,91 @@ def get(self, request, pk, **kwargs):
         ]
         serializer = serializers.CandidateSerializer(candidates, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+class LinkAPIView(CreateAPIView):
+    """
+        View to send the result and vote links if it is an open election
+    """
+    serializer_class = serializers.LinkSerializer
+
+    def create(self, request: Request, *args, **kwargs) -> Response:
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)        
+        id_election = serializer.validated_data["id_election"]
+        select_language = serializer.validated_data["select_language"]
+
+        try:
+            election = Election.objects.get(id=id_election)
+        except Election.DoesNotExist:
+            return Response(
+                WRONG_ELECTION_ERROR,
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        
+        creator_emails = serializer.validated_data.get("creator_emails",[])        
+        
+        if election.on_invitation_only:
+            merge_data: Dict[str, str] = {
+                "result_url": f"{settings.SITE_URL}/result/{election.id}",
+                "title": election.title,
+            }
+
+            activate(
```

Comment:
to avoid code duplication + simplify code visibility, we should put this block as a separate function